tests/tests/test_mtls.py

```diff
@@ -0,0 +1,315 @@
+        deploy.check_expected_statistics(deployment_id, "success", 1)
+        deploy.check_expected_status("finished", deployment_id)
+
+        # verify the update was actually installed on the device
+        out = device.run("mender -show-artifact").strip()
+        assert out == "mtls-artifact"
+
+    @MenderTesting.fast
+    def test_mtls_enterprise_without_client_cert(self, setup_ent_mtls):
+        tenant = setup_ent_mtls.tenant
+
+        # stop the api gateway
+        setup_ent_mtls.stop_api_gateway()
+
+        # start a new mender client
+        setup_ent_mtls.new_mtls_client("mender-client", tenant.tenant_token)
+
+        # stop the Mender client
+        setup_ent_mtls.device = device = MenderDevice(
+            setup_ent_mtls.get_mender_clients()[0]
+        )
+        device.ssh_is_opened()
+        device.run("systemctl stop mender-client")
+        tmpdir = tempfile.mkdtemp()
+        try:
+            # retrieve the original configuration file
+            output = device.run("cat /etc/mender/mender.conf")
+            config = json.loads(output)
+            # replace mender.conf with an mTLS enabled one, without the client certs
+            config["ServerURL"] = "https://mtls-ambassador:8080"
+            config["ServerCertificate"] = "/etc/mender/server.crt"
+            config["SkipVerify"] = True
+            device.run("rm /data/mender/mender.conf")
+            mender_conf = os.path.join(tmpdir, "mender.conf")
+            with open(mender_conf, "w") as fd:
+                json.dump(config, fd)
+            device.put(
+                os.path.basename(mender_conf),
+                local_path=os.path.dirname(mender_conf),
+                remote_path="/etc/mender",
+            )
+        finally:
+            shutil.rmtree(tmpdir)
+
+        # upload the certificates
+        device.put(
+            "tenant.ca.crt",
+            local_path=os.path.join(
+                os.path.dirname(__file__),
+                "..",
+                "..",
+                "extra",
+                "mtls",
+                "certs",
+                "tenant-ca",
+            ),
+            remote_path="/etc/ssl/certs",
+        )
+        device.put(
+            "server.crt",
+            local_path=os.path.join(
+                os.path.dirname(__file__),
+                "..",
+                "..",
+                "extra",
+                "mtls",
+                "certs",
+                "server",
+            ),
+            remote_path="/etc/mender",
+        )
+        device.put(
+            "cert.crt",
+            local_path=os.path.join(
+                os.path.dirname(__file__), "..", "..", "certs", "storage-proxy",
+            ),
+            remote_path="/etc/mender",
+        )
+        device.run(
+            "cat /etc/mender/server.crt /etc/mender/cert.crt > /tmp/server.crt && mv /tmp/server.crt /etc/mender/server.crt && rm /etc/mender/cert.crt"
+        )
```

Comment:
Up until here the code is the same as the previous text; consider refactoring.

Reply:
refactored